bin.js

```diff
@@ -0,0 +1,207 @@
+    runSingle({ benchName, file }, function (err, out) {
+      if (err) {
+        // nothing we can do about it
+        throw err
+      }
+
+      if (outList) {
+        outList.push(out)
+      }
+
+      const f = files.shift()
+
+      if (f) {
+        console.log('')
+        next(f)
+        return
+      } else {
+        write()
+      }
+    })
+  }
+
+  function write () {
+    if (!outList) {
+      return
+    }
+
+    fs.writeFile(args.out, JSON.stringify(outList, null, 2), function (err) {
+      if (err) {
+        throw err
+      }
+    })
+  }
+}
+
+function runSingle (opts, cb) {
+  const benchName = opts.benchName
+  const file = opts.file
+  const silent = opts.silent
+  const args = [
+    file
+  ]
+
+  const env = Object.assign({ BHDR_JSON: true }, process.env)
+  const text = buildText({ color: true })
+
+  const child = childProcess.spawn(process.execPath, args, {
+    cwd: path.join(__dirname),
+    env,
+    stdio: ['ignore', 'pipe', 'pipe'],
+    detached: false
+  })
+
+  child.stderr.pipe(process.stderr)
+  const childOut = child.stdout.pipe(split(function (line) {
+    try {
+      return JSON.parse(line)
+    } catch (err) {
+      return { err: line }
+    }
+  }))
+
+  const out = {
+    name: benchName,
+    experiments: []
+  }
+
+  childOut.pipe(new Transform({
+    objectMode: true,
+    transform: function (chunk, enc, cb) {
+      out.experiments.push(chunk)
+
+      if (!silent) {
+        cb(null, text(chunk) + '\n')
+      } else {
+        cb()
+      }
+    }
+  })).on('end', function () {
+    cb(null, out)
+  })
+  .pipe(process.stdout)
+}
+
+function compare (argv) {
+  const args = minimist(argv)
+
+  if (args._.length !== 2) {
+    // TODO better usage
+    console.error('Usage: bdhr compare old new')
+    process.exit(1)
+  }
+
+  const files = args._.reduce(add, [])
+  const oldBenchName = path.relative(process.cwd(), files[0])
+  const newBenchName = path.relative(process.cwd(), files[1])
+
+  runSingle({
+    benchName: oldBenchName,
+    file: files[0],
+    silent: true
+  }, function (err, outOld) {
+    if (err) {
+      throw err
+    }
+
+    runSingle({
+      benchName: newBenchName,
+      file: files[1],
+      silent: true
+    }, function (err, outNew) {
+      if (err) {
+        throw err
+      }
+
+      const stat = compareData(outOld, outNew)
+
+      const text = buildText({ color: false, prefix: false })
+
+      const data = Object.keys(stat).reduce(function (res, key) {
+        const obj = {}
+
+        obj.bench = key
+        obj[oldBenchName] = text(stat[key].old)
+        obj[newBenchName] = text(stat[key].new)
+
+        const color = stat[key].difference.indexOf('-') === 0 ? chalk.red : chalk.green
+
+        obj.difference = color(stat[key].difference)
+        obj.significant = chalk.bold(stat[key].significant)
+
+        res.push(obj)
+        return res
+      }, [])
+
+      console.log(columnify(data))
+    })
+  })
+}
+
+function add (acc, file) {
```

Comment:
would containing the try/catch in another func help perf or would the extra jmp make it not worth it?